cmake/modules/FindKOKKOS.cmake

```diff
@@ -80,8 +78,8 @@ ELSE()
     LIBRARIES REQUIRED KOKKOS_CORE_LIBRARY KOKKOS_CONTAINERS_LIBRARY
     INCLUDE_DIRS REQUIRED KOKKOS_INSTALL_INCLUDE_DIR
     USER_INCLUDE_DIRS REQUIRED KOKKOS_INSTALL_INCLUDE_DIR
-    CXX_FLAGS OPTIONAL KOKKOS_COMPILE_FLAGS
-    LINKER_FLAGS OPTIONAL KOKKOS_EXTRA_LD_FLAGS
+    CXX_FLAGS OPTIONAL ${KOKKOS_COMPILE_FLAGS}
+    LINKER_FLAGS OPTIONAL ${KOKKOS_EXTRA_LD_FLAGS}
```

Comment:
This needs to be without the enclosing `${...}` - provided both variables, `KOKKOS_COMPILE_FLAGS` and `KOKKOS_EXTRA_LD_FLAGS` contain the required flags.

Reply:
>This needs to be without the enclosing ${...} - provided both variables

Hi @tamiko, I've also tried this one before. The turth is that the `KOKKOS_COMPILE(EXTRA_LD)_FLAGS ` only contains the **string** of the generator expression, and you need to use `${...}` to pass the actual expression which would transform to flags while generating the file cache and make file.

> But then I am not sure any more what the macro itself does.

Have no idea but that works at least :sweat_smile:

Reply:
The macro itself evaluates the string, i.e. looks at `${...}` - this is why I am surprised. I'll try to reproduce early next week.

Would you mind telling me a bit about your setup? (how do you get external kokkos)